File: app/src/main/java/com/pirateman/exercisemanager/interval/IntervalDao.java
package com.pirateman.exercisemanager.interval;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface IntervalDao {

    @Query("select * from interval")
    LiveData<List<Interval>> getAll();

    @Query("select * from interval where id = :id")
    LiveData<Interval> getIntervalById(int id);

    @Query("delete from interval WHERE id = :id")
    void removeInterval(int id);

    @Update
    void updateInterval(Interval interval);

    @Insert
    void insertInterval(Interval interval);

    @Insert
    void insertAll(Interval... intervals);

    @Query("delete from interval")
    void removeAll();

    @Query("SELECT COUNT('id') FROM interval")
    int getNumberOfRows();

    @Query("select * from interval where exerciseId = :exerciseId")
    LiveData<List<Interval>> getIntervalsByExerciseId(int exerciseId);

}

File: app/src/main/java/com/pirateman/exercisemanager/interval/IntervalDao.kt
package com.pirateman.exercisemanager.interval

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface IntervalDao {
    @get:Query("select * from interval")
    val all: LiveData<List<Interval?>?>?

    @Query("select * from interval where id = :id")
    fun getIntervalById(id: Int): LiveData<Interval?>?

    @Query("delete from interval WHERE id = :id")
    fun removeInterval(id: Int)

    @Update
    fun updateInterval(interval: Interval?)

    @Insert
    fun insertInterval(interval: Interval?)

    @Insert
    fun insertAll(vararg intervals: Interval?)

    @Query("delete from interval")
    fun removeAll()

    @get:Query("SELECT COUNT('id') FROM interval")
    val numberOfRows: Int

    @Query("select * from interval where exerciseId = :exerciseId")
    fun getIntervalsByExerciseId(exerciseId: Long): LiveData<List<Interval>>


}